draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1299,28 +1348,25 @@ ciphersuite, respectively, and \<suiteID\> is the Suite ID of the
 encoding used in ciphersuite \<yy\>.
 
 As another example, consider a fictional protocol named Baz that requires
-two independent random oracles, where one oracle outputs points on the curve
-E1 and the other outputs points on the curve E2.
-Reasonable choices of tags for the E1 and E2 oracles are
-"BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-E1" and "BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-E2",
+two independent random oracles to the same curve.
+Reasonable choices of tags for these oracles are
+"BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-ENC1" and "BAZ-V\<xx\>-CS\<yy\>-\<suiteID\>-ENC2",
 respectively, where \<xx\>, \<yy\>, and \<suiteID\> are as described above.
 
 # Utility functions {#utility}
 
 Algorithms in this document make use of utility functions described below.
 
-For security reasons, all field operations, comparisons, and assignments
-MUST be implemented in constant time (i.e., execution time MUST NOT depend
-on the values of the inputs). Guidance on implementing these low-level
-operations in constant time is beyond the scope of this document.
+For security, implementations of these functions SHOULD be constant-time,
+i.e., execution time SHOULD NOT depend on the values of the inputs.
+For such constant-time implementations, all field operations, comparisons, and
+assignments MUST be implemented in constant time.
+Guidance on implementing these low-level operations in constant time is
+beyond the scope of this document.
 
 -   CMOV(a, b, c): If c is False, CMOV returns a, otherwise it returns b.
-    To prevent against timing attacks, this operation must run in constant
-    time, without revealing the value of c.
-    Commonly, implementations assume that the selector c is 1 for True or
-    0 for False. In this case, given a bit string C, the desired selector c can
-    be computed by OR-ing all bits of C together. The resulting selector will be
-    either 0 if all bits of C are zero, or 1 if at least one bit of C is 1.
+    For constant-time implementations, this operation must run in
+    time independent of the value of c.
```

Comment:
We include an examples of how to compute is_square and inv0 in constant time using -- why drop the example here?

Reply:
Ah, I actually didn't realize it was intended as an example. When I read through this it seemed like a low-level implementation detail rather than an example. Also, I'm pretty sure at this point all of the uses of CMOV in the document actually pass in a boolean value, which means it's not necessary to describe the case where there's a bitstring C rather than a boolean c. I can double check this...

Reply:
I went back and checked. Every use of CMOV in this document takes a boolean value as its third operand, so there shouldn't be any need for the extra explanation that I cut here. We could still bring it back if you want, but it does not seem necessary given the rest of the text.

Reply:
👍 Thanks! I think this was leftover from a previous draft where the value was non-binary.